packages/core/src/modules/credentials/v2/V2CredentialPreview.ts

```diff
@@ -0,0 +1,64 @@
+import type { CredentialPreviewOptions } from '../CredentialPreviewAttributes'
+
+import { Expose, Transform, Type } from 'class-transformer'
+import { Equals, IsInstance, ValidateNested } from 'class-validator'
+
+import { JsonTransformer } from '../../../utils/JsonTransformer'
+import { replaceLegacyDidSovPrefix } from '../../../utils/messageType'
+import { CredentialPreviewAttribute } from '../CredentialPreviewAttributes'
+
+/**
+ * Credential preview inner message class.
+ *
+ * This is not a message but an inner object for other messages in this protocol. It is used construct a preview of the data for the credential.
+ *
+ * @see https://github.com/hyperledger/aries-rfcs/blob/master/features/0036-issue-credential/README.md#preview-credential
+ */
+export class V2CredentialPreview {
+  public constructor(options: CredentialPreviewOptions) {
+    if (options) {
+      this.attributes = options.attributes
+    }
+  }
+
+  @Expose({ name: '@type' })
+  @Equals(V2CredentialPreview.type)
+  @Transform(({ value }) => replaceLegacyDidSovPrefix(value), {
+    toClassOnly: true,
+  })
```

Comment:
`replaceLegacyDidSovPrefix` is not needed anymore for v2 messages. This was needed for v1 when we went from `did:sov:xxx` to `https://didcomm.org` prefixes. But the v2 issue credential protocol uses `https://didcomm.org` natively

```suggestion
```